modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -1720,12 +1717,17 @@ private CompletableFuture<Void> stopPartitionForRestartInternal(ZonePartitionId
 
     @TestOnly
     public HybridTimestamp currentSafeTimeForZonePartition(int zoneId, int partId) {
-        return requireNonNull(zoneResourcesManager.getZonePartitionResources(new ZonePartitionId(zoneId, partId))).raftListener()
-                .currentSafeTime();
+        return zonePartitionResources(new ZonePartitionId(zoneId, partId)).raftListener().currentSafeTime();
     }
 
-    @TestOnly
+    /**
+     * Returns resources for the given zone partition.
+     */
     public ZonePartitionResources zonePartitionResources(ZonePartitionId zonePartitionId) {
-        return requireNonNull(zoneResourcesManager.getZonePartitionResources(zonePartitionId));
+        ZonePartitionResources resources = zoneResourcesManager.getZonePartitionResources(zonePartitionId);
+
+        assert resources != null : "Missing resources for zone partition: " + zonePartitionId;
```

Comment:
I prefer JSON-like formatting: `Missing resources for zone partition [zonePartitionId=0_part_12]`.
Just like we use in logging https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=217393104#JavaLoggingGuidelines-MessageFormat